src/metabase/models/metric.clj

```diff
@@ -65,7 +65,7 @@
 
 ;; ## Persistence Functions
 
-(defn create-metric
+(defn create-metric!
```

Comment:
Ideally, yes. Clojure is a functional language and ideally the vast majority of your functions should be pure (i.e., no side effects). Ideally you minimize the number of "dirty" functions with side effects.